### ISSUE TITLE
Aegisub randomly crashes when playing a video on Linux.

### ISSUE THREAD

**Issue by adalbosco** [Aegisub/Aegisub#285]
This seems to be related, or maybe the same bug as #64, reported for MacOS and, as reported there, seems to happen only for longer videos (above 10 minutes or so).  Basically, when you are timing the subtitles or just playing a segment of the audio to type the corresponding line, it sometimes crashes.

I am using Aegisub on Gentoo Linux (ldd output attached).  I also experienced this issue on Arch Linux, although it felt less frequent than what I've seen on Gentoo.

I am also sending the ffprobe output for the video I was subtitling.

The issue seems to be related when Aegisub tries to free part of the audio/video cash, and probably that's why it doesn't happen with shorter videos (no need to free the cash).

Here is where the crash happens:

Thread 1 "aegisub" received signal **SIGBUS**, Bus error.
0x000055555584e06e in DataBlockCache<float, 10, AudioSpectrumCacheBlockFactory>::**Age** (max_size=67108864, this=0x555558705000) at ./aegisub-3.2.2_p20160518-r102/aegisub-3.2.2_p20160518/src/**block_cache.h:152**
```cpp
 137    void Age(size_t max_size)
 138    {  
 139       // Quick way out: get rid of everything
 140       if (max_size == 0)
 141       {
 142          size_t block_count = data.size();
 143          data.clear();
 144          data.resize(block_count);
 145          age.clear();
 146          size = 0;
 147          return;
 148       }
 149       
 150       // Remove old entries until we're under the max size
 151       for (auto it = age.rbegin(); size > max_size && it != age.rend(); )
*152           KillMacroBlock(**it++);
 153    }
```
Attached is complete backtrace.

Backtrace: [aegisub-bt.txt](https://github.com/Aegisub/Aegisub/files/8951948/aegisub-bt.txt)
ldd output: [aegisub-ldd.txt](https://github.com/Aegisub/Aegisub/files/8951949/aegisub-ldd.txt)
fprobe output: [aegisub-ffprobe.txt](https://github.com/Aegisub/Aegisub/files/8951950/aegisub-ffprobe.txt)

**Comment by wangqr** [comment]
Possibly related to:
https://github.com/Aegisub/Aegisub/pull/142
https://github.com/TypesettingTools/Aegisub/issues/33

**Comment by adalbosco** [comment]
> Possibly related to: #142 [TypesettingTools/Aegisub#33](https://github.com/TypesettingTools/Aegisub/issues/33)

Yes, it is the same issue.  I applied the patch you used in the link below and I have been using Aegisub everyday for more than a month now and never ran into this issue again.  So I can attest that it fixes 100% of the problem and doesn't cause any other issues.

https://github.com/wangqr/Aegisub/commit/4dc14eaf2c32bcdb613408919229424b8ea36a22
